Add fw via panorama connectivity to panos connection

shinmog opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
Adding additional functionality to allow the panos connection to connect to a firewall through a Panorama that is managing it.

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Feature Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
panos.py

##### ADDITIONAL INFORMATION
`pandevice` implements firewall via Panorama connectivity as a `firewall.Firewall` child of a `panorama.Panorama`, where the `firewall.Firewall` only has a serial number specified.  This update implements that to support this additional connection option to the user.

shinmog committed:
Add fw via panorama connectivity to panos connection

justjais commented on lib/ansible/module_utils/network/panos/panos.py:
```diff
@@ -95,10 +96,11 @@ def get_pandevice_parent(self, module):
                     _vstr(self.min_pandevice_version)))
 
-        d, host_arg = None, None
+        d, host_arg, serial_number = None, None, None
```
it would be great if you can rename `d` with some meaningful name.

abenokraitis commented:
@shinmog ^^^

shinmog commented:
Just doing a bit more testing to make sure this is the last time I'll have to update the connection code, then will update this PR.

shinmog committed:
updating panos connection as per review comments

shinmog commented:
This update does a few things:

* addresses review comments
* changes the provider `host` param to `ip_address` (as this is how the PAN-OS hostname is currently given to all panos modules)
* fixes a bug with checking for classic provider params (use `.get()` instead of assuming it is there)

shinmog commented:
@abenokraitis

shinmog committed:
Fixing when 'provider' isn't specified, adding deprecation warning when classic provider params are used
